src/hooks/cart.js

```diff
@@ -13,8 +13,12 @@ export default class extends BaseHooks {
     }
 
     itemAdd() {
-        this.$body.on('submit', '[data-cart-item-add]', (event) => {
-            this.emit('cart-item-add', event, event.target);
-        });
+        const elements = document.querySelectorAll('[data-cart-item-add]');
+
+        for (let i = 0; i < elements.length; i++) {
+            elements[i].addEventListener('submit', event => {
```

Comment:
I used the for loop as I heard that it is more performant than using map function.  Article on that https://codeburst.io/javascript-performance-test-for-vs-for-each-vs-map-reduce-filter-find-32c1113f19d7 and dataset https://github.com/dg92/Performance-Analysis-JS.  I also tend to avoid function calls if I can help it as I heard that in general the code would be slightly slower.

Reply:
Sounds good, I also pulled and ran that code against recent node versions and the performance benefits still hold :)

Thank you for sharing the links